elastic/eui
# [Docs] Add support for `.tsx` CodeSandbox demos

### Summary

@cchaos has been gradually converting our various demos/documentation files to Typescript from plain JS. As she discovered in  https://github.com/elastic/eui/pull/5135#discussion_r704715595, adding types to our demo JS is great because it helps us catch type issues/dogfood our own typing - but unfortunately CodeSandbox currently always assumes an `index.js` file and doesn't know how to parse the Typescript syntax:

<img width="1062" alt="" src="https://user-images.githubusercontent.com/549407/132723870-8170b887-c9ad-43f8-8cd7-797573bbcb98.png">

This PR should solve that issue by allowing us to customize sending either `index.js` or `index.tsx`, based on the `source.type` we send in (either `GuideSectionTypes.JS` or `GuideSectionTypes.TSX`).

## QA

- Go to https://eui.elastic.co/pr_5160/#/navigation/tabs
- Scroll down to the Controlled Tabs example, open the Demo JS, and click the CodeSandbox button in the bottom right hand corner
- [x] Confirm the demo opens in CodeSandbox with zero errors and shows `index.tsx` in the file sidebar:

<img width="1374" alt="" src="https://user-images.githubusercontent.com/549407/132725051-bf7365a3-f7f5-45c0-9213-dddff35d5ba7.png">

### Checklist

~- [ ] Check against **all themes** for compatibility in both light and dark modes~
~- [ ] Checked in **mobile**~
~- [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**~
~- [ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#adding-playground-toggles)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md)**~


~- [ ] Added or updated **[jest tests](https://github.com/elastic/eui/blob/master/wiki/testing.md)**~
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
~- [ ] A **[changelog](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~
